packages/client/hmi-client/src/components/home/tera-project-overview-editor.vue

```diff
@@ -0,0 +1,117 @@
+<template>
+	<Editor
+		v-model="editorContent"
+		class="h-full"
+		:class="{ readonly: !hasEditPermission }"
+		:readonly="!hasEditPermission"
+	/>
+</template>
+
+<script setup lang="ts">
+import { ref, onMounted, computed, onUnmounted, watch } from 'vue';
+import Editor from 'primevue/editor';
+import { update } from '@/services/project';
+import { useProjects } from '@/composables/project';
+import { Project } from '@/types/Types';
+
+const AUTO_SAVE_DELAY = 3000;
+const DEFAULT_EDITOR_CONTENT = `
+	<div>
+			<h2>Hey there! </h2>
+			<p>This is your project overview page. Use this space however you like. Not sure where to start? Here are some things you can try:</p>
+	<br>
+			<ul>
+					<li><strong>Upload stuff:</strong> Upload documents, models, code or datasets with the green button in the bottom left corner.</li>
+					<li><strong>Explore and add:</strong> Use the project selector in the top nav to switch to the Explorer where you can find documents, models and datasets that you can add to your project.</li>
+					<li><strong>Build a model:</strong> Create a model that fits just what you need.</li>
+					<li><strong>Create a workflow:</strong> Connect resources with operators so you can focus on the science and not the plumbing.</li>
+			</ul>
+	<br>
+			<p>Feel free to erase this text and make it your own.</p>
+	</div>
+`;
+const props = defineProps<{ project: Project | null }>();
+const editorContent = ref('');
+const lastSavedContent = computed(() => props.project?.overviewContent ?? DEFAULT_EDITOR_CONTENT);
+const hasEditPermission = computed(() =>
+	['creator', 'writer'].includes(props.project?.userPermission ?? '')
+);
+
+let autoSaveIntervalId: NodeJS.Timeout | null = null;
+const startAutoSave = () => {
+	autoSaveIntervalId = setInterval(saveContent, AUTO_SAVE_DELAY);
+};
+const stopAutoSave = () => {
+	if (autoSaveIntervalId === null) return;
+	clearInterval(autoSaveIntervalId);
+	autoSaveIntervalId = null;
+};
+
+const isContentSameAsLastSaved = computed(
+	() =>
+		lastSavedContent.value.length === editorContent.value.length &&
+		lastSavedContent.value === editorContent.value
+);
+const saveContent = async () => {
+	if (!props.project) return;
+	if (isContentSameAsLastSaved.value) return;
+	const res = await update({ ...props.project, overviewContent: editorContent.value });
```

Comment:
Why not use `useProject` to update?

Reply:
`useProject().update` fetches all projects again at the end and we don't need to do that in this page I think.

Reply:
You could update this method to not do that.

Reply:
I'm leaning toward to just use the `update` from the service here since it's simpler for this purpose and has no side effect to deal with.